content/fr/articles/frontend/2016-2-ans-de-react/index.md

```diff
@@ -0,0 +1,232 @@
+
+Pour bien choisir vos packages, regardez l'activité du projet sur GitHub (commits, issues, tests, contributeurs...)
+
+### Moins de code c'est moins de bugs
+
+Votre appli doit utiliser un maximum de code externe, qui est déjà validé, testé... séparément. Publiez un maximum de modules indépendants, en open source si possible, et utilisables hors-contexte ([FIRST principle](https://addyosmani.com/first/)).
+
+Ceci implique de suivre les corrections/évolutions des dits modules et d'utiliser le [semantic versioning](https://docs.npmjs.com/getting-started/semantic-versioning) à bon escient.
+
+
+## Webpack
+
+![](webpack.png)
+
+### Le saviez-vous ?
+
+Webpack a coûté aux devs **2.312.533 heures** en 2015. Par jour.
+
+### Un grand pouvoir implique de grandes responsabilités
+
+Webpack est puissant, il remplace browserify ou vos multiples tâches grunt, gulp, blurp qui géraient votre pipeline d'assets.
+
+À partir du votre point d'entrée de votre application uniquement, par exemple `index.js`, il est capable de servir et bundler toutes les dépendances (code, images, CSS...) de votre projet automatiquement, grâce aux nombreux [loaders](https://webpack.github.io/docs/loaders.html).
+
+Pour cela, les assets doivent être correctement déclarés dans le code :
+
+```js
+// require some CSS
+import styles from './styles.css'
+const ex1 = <div className={ styles.title }>title</div>
+
+// require an image
+import logo from './logo.png'
+const img = <img src={ logo } />
+```
+
+Et la célèbre [config de webpack](https://webpack.github.io/docs/configuration.html) permet de tuner votre build à tous les niveaux.
+
+Plus de détails sur Webpack sur [les articles Putain de code ! dédiés](http://putaindecode.io/fr/articles/js/webpack/premier-exemple/).
+
+## CSS
+
+![css-modules](css-modules.png)
+
+### Le saviez-vous ?
+
+Le CSS est un sport de haut niveau.
+
+### CSS in 2016
+
+Selon la complexité de vos designs, le CSS peut être ce qui prend le plus de temps, entre l'intégration, le responsive, et les animations.
+
+Et la qualité et la modularité du CSS est essentielle pour la maintenabilité des applications.
+
+Pour créer des composants réutilisables nous pouvons maintenant utiliser les [CSS modules](https://github.com/css-modules/css-modules), qui ont l'avantage d'être du CSS classique, mais d'être scopés aux composants, et, accessibles depuis le JavaScript (plus de CSS global). Et le [CSS in JS](https://youtu.be/WyFGfMFjfH4?t=26m23s) peut compléter/remplacer si besoin pour les cas les plus dynamiques.
+
+A défaut, vous pouvez toujours utiliser du CSS global si vous respectez scrupuleusement les [conventions BEM](http://putaindecode.io/fr/articles/css/bem/).
+
+Il existe plusieurs bibliothèques de composants UI assez fournies pour React :
+
+ - [rebass](http://jxnblk.com/rebass/)
+ - [ant.design](http://ant.design)
+ - [semantic-ui](http://semantic-ui.com)
+ - [material-ui](https://github.com/callemall/material-ui)
```

Comment:
material-ui a bcp de problemes, je le remplacerais par react-toolbox http://react-toolbox.com/

Puis tu peux finir avec ce lien http://davidwells.io/19-open-source-react-component-libraries-to-use-in-your-next-project/